Llvm36: Fix and performance improvement

vlj opened the pull request:
The first commit fixes Retro City Rampage. It basically wrap exec_ppu_func_by_index inside a function that catches CPUThreadExit exception, and it propagates it through return values to Decode function which can safely crash since there is no llvm code.
I tried to do a catch-all code with exception_ptr but it didn't work, I suspect support in VS 2015 is still not mature enough atm...

The 2nd, 4th and 5th commit increase performance of llvm recompiler. It allows to reach 60 fps in Retro City Rampage and Metal Slug 3, as long as llvm threshold is set to 10 (and that recompiler thread has recompiled every necessary block).

Nekotekina commented:
Did you look at the recent implementation of SPU recompiler? `std::exception_ptr` works fine for me. Another usage example exists in vm.cpp but unrelated.

Branch needs rebasing.

vlj commented:
Rebased + use exception_ptr. When I first tried I got corrupt pointer error but now it works,
maybe it was because I stored it in a thread_local variable. Or maybe I did something wrong and fixed it inbetween.
Anyways it works now.

danilaml commented:
Wow! Great speed up! Sonic CD starts at 60 fps almost immediately! Even in select menu!

Nekotekina commented on rpcs3/Emu/Cell/PPUThread.cpp:
```diff
@@ -59,7 +59,7 @@ void ppu_decoder_cache_t::initialize(u32 addr, u32 size)
 
 PPUThread::PPUThread(const std::string& name)
-	: CPUThread(CPU_THREAD_PPU, name, WRAP_EXPR(fmt::format("PPU[0x%x] Thread (%s)[0x%08x]", m_id, m_name.c_str(), PC)))
+	: CPUThread(CPU_THREAD_PPU, name, WRAP_EXPR(fmt::format("PPU[0x%x] Thread (%s)[0x%08x]", m_id, m_name.c_str(), PC))), pending_exception(nullptr)
```
Ehm, it's already nullptr by the default constructor...

Nekotekina commented:
I'd clean it a bit, i.e. rename ExecutionStatusPropagateCPUThreadExit

vlj commented:
Rebased + renamed ExecutionStatusPropagateCPUThreadExit and removed nullptr ctor.
